master/usage/policy/secure-metrics.md

```diff
@@ -0,0 +1,415 @@
+   kubectl label pod my-prometheus-pod calico-prometheus-access=true
+   ```
+
+   If you would like to grant access to a specific IP network, you
+   can create a [global network set](../../reference/calicoctl/resources/globalnetworkset) using calicoctl.
+
+   For example, you might want to grant access to your management subnets.
+
+   ```yaml
+   apiVersion: projectcalico.org/v3
+   kind: GlobalNetworkSet
+   metadata:
+     name: calico-prometheus-set
+     labels:
+       calico-prometheus-access: "true"
+   spec:
+     nets:
+     - 172.15.0.0/24
+     - 172.101.0.0/24
+   ```
+
+### Extended procedure for when Typha is enabled
+
+If your {{site.prodname}} installation uses the Kubernetes API datastore and has greater than 50 nodes, it is likely
+that you have installed Typha. This section shows how to use an additional network policy to secure the Typha
+prometheus endpoints.
+
+After following the steps above, create a file named `typha-prometheus-policy.yaml` with the following contents.
+
+```yaml
+# Allow traffic to prometheus only from sources which are
+# labeled as such, but don't impact any other traffic.
+apiVersion: projectcalico.org/v3
+kind: GlobalNetworkPolicy
+metadata:
+  name: restrict-calico-node-prometheus
+spec:
+  # Select all {{site.prodname}} nodes.
+  selector: running-calico == "true"
+  order: 500
+  types:
+  - Ingress
+  ingress:
+  # Deny anything that tries to access the prometheus port
+  # but that doesn't match the necessary selector.
+  - action: Deny
+    protocol: TCP
+    source:
+      notSelector: calico-prometheus-access == "true"
+    destination:
+      ports:
+      - 9093
+```
+
+This policy selects all endpoints which have the label `running-calico: true`, and enforces a single ingress deny rule.
+The ingress rule denies traffic to port 9093 unless the source of traffic has the label `calico-prometheus-access: true`, meaning
+all Calico workload endpoints, host endpoints, and global network sets which do not have the label, as well as any
+other network endpoints unknown to Calico.
+
+Then, use calicoctl to apply this policy.
+
+```shell
+calicoctl apply -f typha-prometheus-policy.yaml
+```
+
+## Using a whitelist approach
+
+This approach denies traffic to and from your hosts by default, and requires that all
+desired communication be explicitly allowed by a network policy. This approach is more secure because
+only explicitly-allowed traffic will get through, but it requires you to know all the ports that should be open on the host
+
+### Overview
+
+The basic process is as follows:
+
+1. Create host endpoints for each node which you'd like to secure.
+1. Create a network policy which allows desired traffic to the {{site.prodname}} metrics endpoints.
+1. Apply labels to allow access to the prometheus metrics.
+
+### Example for calico/node
+
+1. List the nodes on which {{site.prodname}} is running. In this case, we have two nodes in the cluster.
+
+   ```shell
+   $ calicoctl get nodes
```

Comment:
We are trying to move away from this sort of convention and separate the commands from the responses. We also want to move away from using the `$` prefixes. The reason for this is to make it easier for people to copy and paste the commands. At some point, we may add buttons to make the copy even easier. As we move in this direction, it will be good to separate the commands from the responses and add some introductory text to also make this even more clear.